bee-api/bee-rest-api/src/endpoints/routes/api/v1/messages/mod.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    types::{
+        body::{DefaultErrorResponse, ErrorBody, SuccessBody},
+        dtos::MessageDto,
+        responses::{MessageChildrenResponse, MessageMetadataResponse, MessageResponse, SubmitMessageResponse},
+    },
+    AppStorage,
+};
+
+use bee_message::{Message, MessageId};
+use bee_storage::access::{Fetch, Insert};
+
+use axum::{
+    extract::{Extension, Path},
+    response::Json,
+    routing::{get, post},
+    Router,
+};
+use serde_json::{json, Value};
+
+use rand::{prelude::StdRng, Rng, SeedableRng};
+use std::{convert::TryFrom, str::FromStr, sync::Arc};
+
+pub(crate) fn api_routes() -> Router {
+    Router::new()
+        .route("/", post(post_handler))
+        .route("/:messageId", get(get_id_handler))
+        .route("/:messageId/metadata", get(get_id_metadata_handler))
+        .route("/:messageId/raw", get(get_id_raw_handler))
+        .route("/:messageId/children", get(get_id_children_handler))
+}
+
+fn err_to_json(message: String, code: String) -> Json<Value> {
+    Json(json!(ErrorBody::new(DefaultErrorResponse { code, message })))
+}
+
+async fn post_handler(
+    Extension(app_storage): Extension<Arc<AppStorage>>,
+    Json(payload): Json<MessageDto>,
+) -> Json<Value> {
+    let mut rng = StdRng::from_entropy();
+
+    let rand_part1: u128 = rng.gen();
+    let rand_part2: u128 = rng.gen();
+    let rand_id = format!("{:x}{:x}", rand_part1, rand_part2);
```

Comment:
The message ID comes from the message (a hash of it), you're not supposed to put a random message ID